### ISSUE TITLE
Slow behaviour with glob imports

### ISSUE THREAD

**Issue by mvr** [racer-rust/racer#844]
(This could be a non-issue, there's a good chance I'm using `use` wrong.) 

Racer takes 7 seconds on my machine to provide completions in the following situation, even when none of the modules `a` through `i` exist. If you uncomment the last `use`, it takes around a minute.

```rust
use a::*;
use b::*;
use c::*;
use d::*;
use e::*;
use f::*;
use g::*;
use h::*;
// use i::*;

pub fn foo() -> () {
    Whatever::
}
```

**Comment by kngwyu** [comment]
It's interesting and very annoying.
Thanks for reporting.

**Comment by kngwyu** [comment]
Maybe it's because [search_prelude_file](https://github.com/racer-rust/racer/blob/master/src/racer/nameres.rs) runs many times and it seems difficult to fix.
We can accelerate processing by running search_prelude_file just once using regex, but for it, we have to change many codes.

**Comment by mvr** [comment]
When I run with RUST_LOG=debug, it looks like there is something strange and exponential going on in `match_use`: There are many lines like

```
DEBUG 2018-04-27T02:37:13Z: racer::matchers: found a glob: now searching for P[a::c]
DEBUG 2018-04-27T02:37:13Z: racer::matchers: found a glob: now searching for P[d::a]
DEBUG 2018-04-27T02:37:13Z: racer::matchers: found a glob: now searching for P[d::c]
```

in various combinations of `a`-`h`.

**Comment by kngwyu** [comment]
@mvr 
Thanks.
That's interesting.

**Comment by kngwyu** [comment]
Ah I understand.
When racer tries to find the definition of `a`, it searches `b::a` for the situation like this.
```rust
mod b {
    pub mod a {}
}
use a::*;
use b::*;
```

So I think we can reduce this kind of complexity by caching the result of `resolve_path`.

**Comment by kngwyu** [comment]
Closed via #875 (though it's still a bit slow)

**Comment by mvr** [comment]
Hopefully this was the problem with #803 and #757 too!
